src/nickel.rs

```diff
@@ -156,7 +157,15 @@ impl<D: Sync + Send + 'static> Nickel<D> {
         });
 
         let server = Server::new(self.middleware_stack, self.data);
-        let listener = server.serve(addr, self.keep_alive_timeout).unwrap();
+
+        let is_test_harness = env::vars().any(|(ref k, _)| k == "NICKEL_TEST_HARNESS");
```

Comment:
This is probably the part that feels like the biggest smell.

Alternatives: 
- change all the examples to run on a random port (which is what I did in similar tests for nickel-mustache). The downside of that approach is that users who copy the examples have to actually pay attention to the port (and it can change between server reboots).
- use something more standard such as `$PORT` if it exists, which might be beneficial for users who want to deploy on something like Heroku (I certainly have a tendency to always forget about setting this)
- update examples to read their port from stdin. This is potentially something the user would add themselves, which would help them, but can add some verbosity to examples.